### ISSUE TITLE
 entitlements is null

### ISSUE THREAD

**Issue by stone-4827321** [ealeksandrov/ProvisionQL#46]
the entitlements for ipa is null but use 'codesign -d path --ent -' can show entitlements info

**Comment by stone-4827321** [comment]
the 'codesign -d path --ent -:' is unuseful

**Comment by ealeksandrov** [comment]
Hi! Thanks for report! Can you please send me ipa file where we can reproduce the issue?
You can use email from my GitHub profile.

**Comment by ealeksandrov** [comment]
Thank you for the example `ipa`!
After a closer look, I found that in your case `codesign` fails with this error:
```
Warning: binary contains an invalid entitlements blob. The OS will ignore these entitlements.
```

If we remove the `:` parameter (or newer `--xml` variant), we can see keys and values, but:
- we can't easily parse and print them
- the error above may be more important than just showing the underlying data.

I've found discussions about the ITMS-90164 error that occurs when you have invalid data in entitlements, so it's valuable that we can show failed validation early.

So I've decided to keep the current behaviour, but add detailed error output. Update pushed in [release 1.6.4](https://github.com/ealeksandrov/ProvisionQL/releases/tag/1.6.4).
